app/src/components/market/steps/items/outcomes_step.tsx

```diff
@@ -29,17 +48,54 @@ const OutcomesStep = (props: Props) => {
   const { handleOutcomesChange, values } = props
   const { question, outcomes } = values
 
+  const totalOutcomesWithEmptyNames = outcomes.reduce(
+    (emptyOutcomes: Outcome[] = [], cur: Outcome) => {
+      if (!cur.name) emptyOutcomes.push(cur)
+      return emptyOutcomes
+    },
+    [],
+  ).length
+  const messageOutcomeEmptyNamesError =
+    totalOutcomesWithEmptyNames > 0 ? 'The names of the outcomes should not be empty.' : ''
+
+  const totalOutcomeProbabilities = outcomes.reduce((prev, cur) => prev + cur.probability, 0)
+  const messageOutcomeProbabilitiesError =
+    totalOutcomeProbabilities !== 100 ? 'The sum of all probabilities must be equal to 100%.' : ''
+
+  const error = totalOutcomeProbabilities !== 100 || totalOutcomesWithEmptyNames > 0
+
+  const isAddNewOutcomeButtonDisabled = outcomes.length >= MAX_OUTCOME_ALLOWED
+
+  const addNewOutcome = () => {
+    const newOutcome = {
+      name: '',
+      probability: 0,
+    }
+    outcomes.push(newOutcome)
+    handleOutcomesChange(outcomes)
+  }
+
   return (
     <CreateCard>
       <OutcomeInfo>
         Please add all the possible outcomes for the <strong>&quot;{question}&quot;</strong>{' '}
         question.
       </OutcomeInfo>
-      <Outcomes outcomes={outcomes} onChange={handleOutcomesChange} />
-      <ButtonContainer>
+      <Outcomes
+        outcomes={outcomes}
+        onChange={handleOutcomesChange}
+        messageEmptyNamesError={messageOutcomeEmptyNamesError}
+        messageProbabilitiesError={messageOutcomeProbabilitiesError}
```

Comment:
I'm fine with this as long as there are only two props for it. If at some point we need to add a third possible error condition, we'll need to think of a different approach.